blog-posts/getting-started-gatsby-next-remix.md

```diff
@@ -0,0 +1,306 @@
+  const data = await res.json()
+
+  // Pass data to the page via props
+  return { props: { data } }
+}
+
+export default Page
+```
+
+#### **Remix**
+
+Remix has an exported `loader` function in which the data (e.g. local markdown file) is loaded (similar to `getStaticProps` in Next.js). This `loader` function can receive a `{params}` object. `{params}` can be used to access the path with `params.slug` and load the corresponding data. If you use the `params` object and want to build the route depending on the data, it is important to name your file with a `$` character and the key of the pathname. So in our example, it is `$slug.tsx`.
+In the page component (here `PostSlug`) the data can then be accessed via the `useLoader` hook.
+
+```tsx
+export const loader: LoaderFunction = async ({params}) => {
+    invariant(params.slug, "expected params.slug");
+    return getPost(params.slug);
+};
+
+export default function PostSlug() {
+    const post = useLoaderData();
+    return (
+            <div dangerouslySetInnerHTML={{ __html: post.html }} />
+    );
+}
+```
+
+## Adding Page Metadata
+
+The page meta data, such as the title or description, are especially important for SEO. In the following examples, a title should be added to the pages. For this, each framework offers its own solution.
+
+#### **Gatsby**
+
+To add page metadata in Gatsby, the `gatsby-plugin-react-helmet` and `React Helmet` are recommended. Generally, Gatsby plugins are installed via a package manager and then have to be added to `gatsby-config.js`.
+So in our example blog we first had to run `npm install gatsby-plugin-react-helmet react-helmet` and then add the plugin to `gatsby-config.js`:
+
+```jsx
+{
+  plugins: [`gatsby-plugin-react-helmet`]
+}
+```
+
+After that, you can access the `React Helemet` within the JSX of a component, like for example in the index page.
+
+```tsx
+//...
+import { Helmet } from "react-helmet"
+
+const Index = () => {
+    return (
+        <Layout>
+            <Helmet>
+                <title>Gatsby Blog</title>
+            </Helmet>
+						{/*content goes here*/}
+        </Layout>
+    );
+}
+```
+
+#### **Next.js**
+
+Next.js provides a `head` component that can be built into any page to add metadata.
+
+```tsx
+//...
+import Head from "next/head";
+
+const Home: NextPage = () => {
+    return (
+        <Layout>
+            <Head>
+                <title>Next.js Blog</title>
+								{/*content goes here*/}
+            </Head>
+        </Layout>
+    );
+}
+```
+
+#### **Remix**
+
+To add metadata to Remix projects you have to add the `<Meta />` component in the `<head>` part of `root.tsx`. After that, a `meta` function can be exported to other pages. This must return an object with all relevant metadata. In our example, this is only the page `title`.
+
+```tsx
+//...
+import {MetaFunction} from "remix";
+
+export const meta: MetaFunction = () => {
+    return {
+        title: "Remix Blog",
+    };
+};
```

Comment:
```
import type { MetaFunction } from "remix";

export const meta: MetaFunction = () => {
  return {
    title: "Josie's Shake Shack", // <title>Josie's Shake Shack</title>
    description: "Delicious shakes", // <meta name="description" content="Delicious shakes">
    "og:image": "https://josiesshakeshack.com/logo.jpg" // <meta property="og:image" content="https://josiesshakeshack.com/logo.jpg">
  };
};
```

This is the example from the Remix documentation. Would you like to see a more detailed example in the blogpost than just the html title? My thought behind this was to keep the examples as simple and understandable as possible.

Reply:
oh that's surprising. They map description to meta description (name notation) and title to the html title while the og tag is mapped to the meta `property` notation. This is quite some magic happening here isn't it ? I think that example is much better. 

A simple title property can't tell me how on earth it would allow me to create meta tags. Especially compared to the straight forward and magic-less approaches from the other frameworks.

Reply:
Yes definitely makes sense. I have now added `description` and `og:image` to the examples and explained this in a bit more detail in Remix

Reply:
There is very little magic involved: https://remix.run/docs/en/v1/api/conventions#meta

Reply:
Thanks for the changes. Looks much better.